### DISCUSSION TITLE
How do I log a message to the console in C++

### DISCUSSION THREAD

**Discussion by HydrogenC** [o3de/o3de#17061] [Q&A] [answered]
I browzed the user guide for half an hour but failed to find how to log a message to the editor console in C++.

**Comment by anon-oss** [comment, marked as answer]
// message
AZ_Printf("UiManagerClass", "loading canvas");

// message with string
AZ_Printf("InventoryComponentClass", " New Item Name = %s", Items[inItemLocation].c_str());

// message with float
AZ_Printf("time delta", "deltatime %f: time %f", dt, Totaltime);

It might be good to start with O3DE by going through book "Game Programming with Open 3D Engine" by Oleksandr Lozitskiy. Its available on amazon and also freely available at his repository https://github.com/AMZN-Olex/O3DEBookCode2111
